plaso/engine/configurations.py

```diff
@@ -211,6 +212,8 @@ class ProcessingConfiguration(interface.AttributeContainer):
   def __init__(self):
     """Initializes a process configuration object."""
     super(ProcessingConfiguration, self).__init__()
+    self.artifacts_registry = None
```

Comment:
Re: artifacts_registry

Please don't pass complex objects via configuration configuration.artifacts_registry. Configuration is meant to pass settings. If you need pass complex objects do this via the methods.

also it is not defined in the docstring.

Reply:
It seems we were storing a complete ArtifactDefinitionsRegistry() in the configuration object, prior to these changes.  However, I don't think it was ever used, so I've refactored the helper to store the artifact definitions path and customer artifact definitions path only.  All tests are passing after refactoring, so I think I've accounted for any other impact.